### PR TITLE
Support devcontainer (local and GH Codespaces)

### PR DESCRIPTION
**Pull request by tfenster**
As an alternative to locally installing the toolset, you can also use a devcontainer. This PR adds the required files and documentation how to use it.

### CODE REVIEW

**Comment by tfenster** [comment]
@waldo1001 @JeremyVyska as discussed today :)

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Update the NODE_VERSION arg in docker-compose.yml to pick a Node version: 10, 12, 14
+ARG NODE_VERSION=14
+FROM mcr.microsoft.com/vscode/devcontainers/javascript-node:0-${NODE_VERSION}
+
+# VARIANT can be either 'hugo' for the standard version or 'hugo_extended' for the extended version.
+ARG VARIANT=hugo
```
Comment (TheDoubleH) [inline .devcontainer/Dockerfile:6]: Our theme requires the extended version.

See: https://www.docsy.dev/docs/getting-started/

```suggestion
ARG VARIANT=hugo_extended
```

File: content/docs/Contributing/InstallHugo/TheShortcut/index.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+---
+Title: Devcontainer with VS Code
+---
+
+#### Use a local devcontainer
```
Comment (TheDoubleH) [inline content/docs/Contributing/InstallHugo/TheShortcut/index.md:5]:
```suggestion
## Use a local devcontainer
```

File: content/docs/Contributing/InstallHugo/TheShortcut/index.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+8. Now you can make changes and just save them. If you open the terminal, you will see a message that tells you that a change was detected and the site was rebuilt. After that, the change should automatically appear in your browser
+
+Here is a walkthrough of the full process:
+
+<video width=100% controls>
+    <source src="alguidelines walkthrough.mp4" type="video/mp4">
+    Your browser does not support the video tag.  
+</video>
+
+#### Use GitHub Codespaces
```
Comment (TheDoubleH) [inline content/docs/Contributing/InstallHugo/TheShortcut/index.md:25]:
```suggestion
## Use GitHub Codespaces
```

**Review by TheDoubleH — CHANGES_REQUESTED** [review]
- Request to use extended version of hugo
- Headline

File: content/docs/Contributing/InstallHugo/TheShortcut/index.md (hunk 2 above)
Comment (tfenster) [reply content/docs/Contributing/InstallHugo/TheShortcut/index.md]: Fixed, thanks

File: content/docs/Contributing/InstallHugo/TheShortcut/index.md (hunk 3 above)
Comment (tfenster) [reply content/docs/Contributing/InstallHugo/TheShortcut/index.md]: Fixed, thanks

**Review by TheDoubleH — APPROVED** [review]

**Comment by TheDoubleH** [comment]
@tfenster  / @ajkauffmann 

Might be an idea to add a warning about that.

```
{{% alert title="Warning" color="warning" %}}
Blah Blah Blah
{{% /alert %}}

**Comment by tfenster** [comment]
> If you can add a message about to not be concerned if / when you get that.

@TheDoubleH  I added something, see https://github.com/microsoft/alguidelines/pull/143/commits/cda4104e6189611866a03e0ab0c31d751712e411. Does that make sense and helps to avoid the concerns? Or is it too much?

**Comment by tfenster** [comment]
@waldo1001 as discussed: https://github.com/microsoft/alguidelines/pull/143/commits/408368b3de51e324d78c874d51761f606e33a6e2

**Review by ajkauffmann — APPROVED** [review]

**Pull request merged**